src/components/MoneyRequestConfirmationList.js

```diff
@@ -100,260 +100,233 @@ const defaultProps = {
         });
         return [
             {
                 text: text[0].toUpperCase() + text.slice(1),
-                value: this.props.hasMultipleParticipants ? CONST.IOU.MONEY_REQUEST_TYPE.SPLIT : CONST.IOU.MONEY_REQUEST_TYPE.REQUEST,
+                value: props.hasMultipleParticipants ? CONST.IOU.MONEY_REQUEST_TYPE.SPLIT : CONST.IOU.MONEY_REQUEST_TYPE.REQUEST,
             },
         ];
-    }
+    }, [props.hasMultipleParticipants, props.iouAmount, props.iou.selectedCurrencyCode, translate]);
 
-    /**
-     * Get selected participants
-     * @returns {Array}
-     */
-    getSelectedParticipants() {
-        return _.filter(this.state.participants, (participant) => participant.selected);
-    }
-
-    /**
-     * Get unselected participants
-     * @returns {Array}
-     */
-    getUnselectedParticipants() {
-        return _.filter(this.state.participants, (participant) => !participant.selected);
-    }
-
-    /**
-     * Returns the participants with amount
-     * @param {Array} participants
-     * @returns {Array}
-     */
-    getParticipantsWithAmount(participants) {
-        const iouAmount = IOUUtils.calculateAmount(participants.length, this.props.iouAmount);
-        return OptionsListUtils.getIOUConfirmationOptionsFromParticipants(participants, CurrencyUtils.convertToDisplayString(iouAmount, this.props.iou.selectedCurrencyCode));
-    }
+    const selectedParticipants = useMemo(() => _.filter(participants, (participant) => participant.selected), [participants]);
 
     /**
      * Returns the participants without amount
      *
      * @param {Array} participants
      * @returns {Array}
      */
-    getParticipantsWithoutAmount(participants) {
-        return _.map(participants, (option) => _.omit(option, 'descriptiveText'));
-    }
+    const getParticipantsWithoutAmount = useCallback((participantsList) => _.map(participantsList, (option) => _.omit(option, 'descriptiveText')), []);
 
-    /**
-     * Returns the personalDetails object for the payee. Use the payee prop if passed, else fallback to current user
-     *
-     * @returns {Object} personalDetails
-     */
-    getPayeePersonalDetails() {
-        return this.props.payeePersonalDetails || this.props.currentUserPersonalDetails;
-    }
+    const payeePersonalDetails = useMemo(() => props.payeePersonalDetails || props.currentUserPersonalDetails, [props.payeePersonalDetails, props.currentUserPersonalDetails]);
 
-    /**
-     * Returns the sections needed for the OptionsSelector
-     *
-     * @returns {Array}
-     */
-    getSections() {
+    const optionSelectorSections = useMemo(() => {
         const sections = [];
-        if (this.props.hasMultipleParticipants) {
-            const selectedParticipants = this.getSelectedParticipants();
-            const unselectedParticipants = this.getUnselectedParticipants();
+        const unselectedParticipants = _.filter(participants, (participant) => !participant.selected);
```

Comment:
Don't we need to use `useMemo` here?

Reply:
This is called inside `optionSelectorSections` which is already memoized.